### PR TITLE
:sparkles: improving error handling for lazy-loading

### PR DESCRIPTION
**Pull request by MatissJanis**
Added a custom error message if lazy-loading fails.

Screenshot 2024-04-23 at 17 55 32

### CODE REVIEW

File: packages/desktop-client/src/components/util/LoadComponent.tsx [hunk 1]
```diff
@@ -19,22 +21,36 @@ export function LoadComponent<K extends string>(props: LoadComponentProps<K>) {
   return <LoadComponentInner key={props.name} {...props} />;
 }
 
+// Cache of the various modules so we would not need to
+// load the same thing multiple times.
+const localModuleCache = new Map();
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/util/LoadComponent.tsx:26]: Oh this should also make the subsequent loads _slightly_ faster.

File: packages/desktop-client/src/components/FatalError.tsx [hunk 2]
```diff
@@ -157,7 +173,7 @@ export function FatalError({ buttonText, error }: FatalErrorProps) {
   const showSimpleRender = 'type' in error && error.type === 'app-init-failure';
 
   return (
-    <Modal isCurrent={true} CloseButton={undefined} title="Fatal Error">
+    <Modal isCurrent title="Fatal Error">
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/FatalError.tsx:176]: Maybe we can accept a custom title here and pass in "Loading Error" for lazy loading errors.

Comment (MatissJanis) [reply packages/desktop-client/src/components/FatalError.tsx]: Sure: https://github.com/actualbudget/actual/pull/2657/commits/6dd311f12fcc9e5d166e3f1d35d10c00ea590a19

It required a small amount of refactoring though. :)

**Comment by joel-jeremy** [comment]
Opened the demo link and tried creating a test budget but got errors. The loading screen doesn't seem to show as well.

**Comment by MatissJanis** [comment]
How can I repro the errors you are seeing?

The loading indicator when loading the budget is not a lazy-loading issue I think. There's different logic for that, so it's unrelated to this PR.

**Review by joel-jeremy — APPROVED** [review]
The loading were just exhausting all the retries on my end for some reason but that could just be my computer acting up. I'm good with merging this PR. :)

**Pull request merged**